libqtile/widget/statusnotifier.py

```diff
@@ -508,6 +512,14 @@ def item_added(self, item, service, future):
             if self.on_item_added:
                 self.on_item_added(item)
 
+        # It's an invalid item so let's remove it from the watchers
+        else:
+            for w in self.watchers:
+                try:
+                    w._items.remove(service)
+                except ValueError:
+                    pass
```

Comment:
AFAICT `StatusNotifierHost` has a `watchers` attribute, but `StatusNotifierItem` here doesn't. Is this right?

Reply:
This is in `StatusNotifierHost`.

Reply:
Oh, lol. github and its code folding. Nevermind!

Reply:
That's caught me out more times than I'd like to admit!